lib/pure/logging.nim

```diff
@@ -87,12 +92,13 @@ type
 {.deprecated: [TLevel: Level, PLogger: Logger, PConsoleLogger: ConsoleLogger,
     PFileLogger: FileLogger, PRollingFileLogger: RollingFileLogger].}
 
-proc substituteLog(frmt: string): string =
-  ## converts $date to the current date
-  ## converts $time to the current time
-  ## converts $app to getAppFilename()
-  ## converts
-  result = newStringOfCap(frmt.len + 20)
+proc substituteLog(frmt: string, level: Level, args: varargs[string, `$`]): string =
+  var msg = ""
```

Comment:
The point is to get rid of this temporary 'msg' string and the allocations it causes. ;-)